main.tf

```
@@ -62,7 +62,7 @@ resource "aws_security_group" "default" {
 module "dns" {
   source    = "git::https://github.com/cloudposse/terraform-aws-route53-cluster-hostname.git?ref=tags/0.2.5"
   enabled   = "${local.enabled && length(var.zone_id) > 0 ? "true" : "false"}"
-  name      = "${module.label.id}"
+  name      = "${var.dns_name == "" ? module.label.id : "${var.dns_name}"}"
```

```suggestion
  name      = "${var.dns_name == "" ? module.label.id : var.dns_name}"
```

No need to use interpolation inside of interpolation

---

Fixed.